src/coreComponents/physicsSolvers/fluidFlow/ThermalCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -0,0 +1,685 @@
+      for( integer ke = 0; ke < stack.stencilSize; ++ke )
+      {
+        dPhaseFlux_dT[ke] += dPresGrad_dT[ke];
+      }
+      for( integer ke = 0; ke < stack.numFluxElems; ++ke )
+      {
+        dPhaseFlux_dT[ke] -= dGravHead_dT[ke];
+      }
+      for( integer ke = 0; ke < stack.stencilSize; ++ke )
+      {
+        dPhaseFlux_dT[ke] *= m_phaseMob[er_up][esr_up][ei_up][ip];
+      }
+      dPhaseFlux_dT[k_up] += m_dPhaseMob_dTemp[er_up][esr_up][ei_up][ip] * potGrad;
+
+      // Step 3.2: compute the derivative of component flux wrt temperature
+
+      // slice some constitutive arrays to avoid too much indexing in component loop
+      arraySlice1d< real64 const, multifluid::USD_PHASE_COMP - 3 > phaseCompFracSub =
+        m_phaseCompFrac[er_up][esr_up][ei_up][0][ip];
+      arraySlice2d< real64 const, multifluid::USD_PHASE_COMP_DC - 3 > dPhaseCompFracSub =
+        m_dPhaseCompFrac[er_up][esr_up][ei_up][0][ip];
+
+      for( integer ic = 0; ic < numComp; ++ic )
+      {
+        real64 const ycp = phaseCompFracSub[ic];
+        for( integer ke = 0; ke < stack.stencilSize; ++ke )
+        {
+          dCompFlux_dTSlice[ke][ic] += dPhaseFlux_dT[ke] * ycp;
+        }
+        dCompFlux_dTSlice[k_up][ic] += phaseFlux * dPhaseCompFracSub[ic][Deriv::dT];
+      }
+
+      // Step 4: compute the enthalpy flux
+
+      real64 const enthalpy = m_phaseEnthalpy[er_up][esr_up][ei_up][0][ip];
+      *energyFluxPtr += phaseFlux * enthalpy;
+
+      for( integer ke = 0; ke < stack.stencilSize; ++ke )
+      {
+        dEnergyFlux_dPSlice[ke] += dPhaseFlux_dP[ke] * enthalpy;
+        dEnergyFlux_dTSlice[ke] += dPhaseFlux_dT[ke] * enthalpy;
+
+        for( integer jc = 0; jc < numComp; ++jc )
+        {
+          dEnergyFlux_dCSlice[ke][jc] += dPhaseFlux_dC[ke][jc] * enthalpy;
+        }
+      }
+
+      dEnergyFlux_dPSlice[k_up] += phaseFlux * m_dPhaseEnthalpy[er_up][esr_up][ei_up][0][ip][Deriv::dP];
+      dEnergyFlux_dTSlice[k_up] += phaseFlux * m_dPhaseEnthalpy[er_up][esr_up][ei_up][0][ip][Deriv::dT];
+
+      real64 dProp_dC[numComp]{};
+      applyChainRule( numComp,
+                      m_dCompFrac_dCompDens[er_up][esr_up][ei_up],
+                      m_dPhaseEnthalpy[er_up][esr_up][ei_up][0][ip],
+                      dProp_dC,
+                      Deriv::dC );
+      for( integer jc = 0; jc < numComp; ++jc )
+      {
+        dEnergyFlux_dCSlice[k_up][jc] += phaseFlux * dProp_dC[jc];
+      }
+    } );
+
+    // *****************************************************
+    // Computation of the conduction term in the energy flux
+    // Note that this term is computed using an explicit treatment of conductivity for now
+    // Step 1: compute the thermal transmissibilities at this face
+    m_stencilWrapper.computeWeights( iconn,
+                                     m_thermalConductivity,
+                                     m_thermalConductivity, // unused for now
```

Comment:
why twice? I m guessing it's just to pass smt to the function coz thta should be the derivative wrt to pressure.

Reply:
For thermal conductivity, I implemented what the two of us discussed some time ago: the dependence of thermal conductivity on saturation and porosity is treated explicitly to make sure that we can easily reuse the `computeWeights` function. So yes you are right, I pass `m_thermalConductivity` twice because I need to pass something, but I don't have anything else to pass. I added more explanations in the code.

In a future PR, we could modify the `computeWeights` function to accommodate explicit coefficients  (just pass the coefficient, and return the transmissibility). That would be useful for thermal transmissibility and also for diffusion transmissibility that would likely be explicit/lagged in iteration as well.